### PR TITLE
Fix documentation issues regarding range selection

### PR DESCRIPTION
**Pull request by hegyibalint**

### CODE REVIEW

**Comment by hegyibalint** [comment]
@bot-gradle RFM test

**Review by ljacomet — CHANGES_REQUESTED** [review]
Nice added test and description!

A couple suggestions:

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/VersionConflictResolutionIntegrationTest.groovy [hunk 1]
```diff
@@ -1075,6 +1075,39 @@ task checkDeps(dependsOn: configurations.compile) {
...
+            configurations {
+                conf
+            }
+            dependencies {
+                conf 'org:a:1.0', 'org:b:1.0'
+            }
+            task checkDeps {
+                doLast {
+                    def files = configurations.conf*.name.sort()
+                    assert files == ['a-1.0.jar', 'b-1.0.jar', 'leaf-5.jar']
```
[the remark this answers is not in the document]
Comment (hegyibalint) [reply subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/VersionConflictResolutionIntegrationTest.groovy:1099]: Ah, I see - we are going to fail before that!

File: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_resolution.adoc [hunk 2]
```diff
@@ -77,7 +77,9 @@ As you have seen, Gradle supports a concept of <<rich_versions.adoc#rich-version
 ** If a version declared as `strictly` is lower than that version, selection will fail.
 * If ranges are involved:
 ** If there is a non range version that falls within the specified ranges or is higher than their upper bound, it will be selected.
-** If there are only ranges, the highest _existing_ version of the range with the highest upper bound will be selected.
+** If there are only ranges, the selection will depend on the intersection of ranges:
+*** If all the ranges intersect, then the highest _existing_ version of the intersection will be selected.
+*** If there is no clear intersection between all the ranges, the highest _existing_ version will be selected from the highest range. If there is no artifact available for the highest range, the resolution will fail.
```
[the remark this answers is not in the document]
Comment (hegyibalint) [reply subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_resolution.adoc:82]: Regarding the text:
 - I've tried to apply the same style as before, but it's true that _existing_ is a little bit of a noise here.
 - Should we remove _existing_ in both cases to simplify the statement?

Regarding the logic:
 - This branch is the "If there are only ranges" one, so it should be certain that there are no fixed versions defined.
 - I would think that in this case, Gradle would list the elements, as there is no other choice.

[the remark this answers is not in the document]
Comment (ljacomet) [reply subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_resolution.adoc:82]: About removing _existing_ in the previous sentence: might be worth checking what happens with [3,6] and [4,7] when 6 is not an existing version.

For the second point, what I mean is that you suddenly use the word "artifact" which does not appear elsewhere. I think you should use version instead.

[the remark this answers is not in the document]
Comment (hegyibalint) [reply subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_resolution.adoc:82]: > About removing existing in the previous sentence: might be worth checking what happens with [3,6] and [4,7] when 6 is not an existing version.
I will definitely check this out, as I haven't found tests for these corner cases.

> For the second point, what I mean is that you suddenly use the word "artifact" which does not appear elsewhere. I think you should use version instead.
Ah, I see... Who said that we are looking for an _artifact_, we are just hunting for information. :laughing:

**Review by ljacomet — APPROVED** [review]

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Pull request merged**